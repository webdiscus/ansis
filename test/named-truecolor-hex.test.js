import { describe, test, expect, vi } from 'vitest';

import colorNames from 'css-color-names';
import ansis, { Ansis } from 'ansis';

// truecolor level
const level = 3;

// mock ansis to ensure it always uses the required colorNames level,
// independent of what would be auto-detected
vi.mock('ansis', async () => {
  const { Ansis } = await vi.importActual('ansis');
  const ansis = new Ansis(3); // force init with level
  return { default: ansis, Ansis }
});

describe('use colorNames name with ansis.hex()', () => {
  test('simple', async () => {
    // colorNames.pink => #ffc0cb
    const received = ansis.hex(colorNames.pink)('Pink via HEX');
    const expected = '[38;2;255;192;203mPink via HEX[39m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('helper', async () => {
    const truecolor = (name) => {
      const hex = colorNames[name.toLowerCase()];
      if (!hex) throw new Error(`unknown color: ${name}`);
      return ansis.hex(hex);
    };

    const received = truecolor('Orange').underline('Orange via HEX');
    const expected = '[38;2;255;165;0m[4mOrange via HEX[24m[39m';

    console.log(received);
    expect(received).toEqual(expected);
  });

  test('single extended colorNames via new instance', async () => {
    const ansis = new Ansis(level).extend(colorNames);
    const received = ansis.pink('Pink');
    const expected = '[38;2;255;192;203mPink[39m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('single extended colorNames', async () => {
    const color = ansis.extend(colorNames);
    const received = color.pink('Pink');
    const expected = '[38;2;255;192;203mPink[39m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('single styled extended colorNames', async () => {
    const ansis = new Ansis(level).extend(colorNames);
    const received = ansis.pink.underline('Pink');
    const expected = '[38;2;255;192;203m[4mPink[24m[39m';
    console.log(received);
    expect(received).toEqual(expected);
  });

  test('print all named truecolor', async () => {
    const ansis = new Ansis(level).extend(colorNames);
    const names = Object.keys(colorNames);
    const received = names.map((name) => ansis[name](name)).join('\n');

    console.log(received);

    expect(received).toMatchSnapshot();
  });
});