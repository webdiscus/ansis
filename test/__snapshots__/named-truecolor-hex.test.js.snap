// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`use color name with ansis.hex() > print all named truecolor 1`] = `
"[38;2;240;248;255maliceblue[39m
[38;2;250;235;215mantiquewhite[39m
[38;2;0;255;255maqua[39m
[38;2;127;255;212maquamarine[39m
[38;2;240;255;255mazure[39m
[38;2;245;245;220mbeige[39m
[38;2;255;228;196mbisque[39m
[38;2;0;0;0mblack[39m
[38;2;255;235;205mblanchedalmond[39m
[38;2;0;0;255mblue[39m
[38;2;138;43;226mblueviolet[39m
[38;2;165;42;42mbrown[39m
[38;2;222;184;135mburlywood[39m
[38;2;95;158;160mcadetblue[39m
[38;2;127;255;0mchartreuse[39m
[38;2;210;105;30mchocolate[39m
[38;2;255;127;80mcoral[39m
[38;2;100;149;237mcornflowerblue[39m
[38;2;255;248;220mcornsilk[39m
[38;2;220;20;60mcrimson[39m
[38;2;0;255;255mcyan[39m
[38;2;0;0;139mdarkblue[39m
[38;2;0;139;139mdarkcyan[39m
[38;2;184;134;11mdarkgoldenrod[39m
[38;2;169;169;169mdarkgray[39m
[38;2;0;100;0mdarkgreen[39m
[38;2;169;169;169mdarkgrey[39m
[38;2;189;183;107mdarkkhaki[39m
[38;2;139;0;139mdarkmagenta[39m
[38;2;85;107;47mdarkolivegreen[39m
[38;2;255;140;0mdarkorange[39m
[38;2;153;50;204mdarkorchid[39m
[38;2;139;0;0mdarkred[39m
[38;2;233;150;122mdarksalmon[39m
[38;2;143;188;143mdarkseagreen[39m
[38;2;72;61;139mdarkslateblue[39m
[38;2;47;79;79mdarkslategray[39m
[38;2;47;79;79mdarkslategrey[39m
[38;2;0;206;209mdarkturquoise[39m
[38;2;148;0;211mdarkviolet[39m
[38;2;255;20;147mdeeppink[39m
[38;2;0;191;255mdeepskyblue[39m
[38;2;105;105;105mdimgray[39m
[38;2;105;105;105mdimgrey[39m
[38;2;30;144;255mdodgerblue[39m
[38;2;178;34;34mfirebrick[39m
[38;2;255;250;240mfloralwhite[39m
[38;2;34;139;34mforestgreen[39m
[38;2;255;0;255mfuchsia[39m
[38;2;220;220;220mgainsboro[39m
[38;2;248;248;255mghostwhite[39m
[38;2;218;165;32mgoldenrod[39m
[38;2;255;215;0mgold[39m
[38;2;128;128;128mgray[39m
[38;2;0;128;0mgreen[39m
[38;2;173;255;47mgreenyellow[39m
[38;2;128;128;128mgrey[39m
[38;2;240;255;240mhoneydew[39m
[38;2;255;105;180mhotpink[39m
[38;2;205;92;92mindianred[39m
[38;2;75;0;130mindigo[39m
[38;2;255;255;240mivory[39m
[38;2;240;230;140mkhaki[39m
[38;2;255;240;245mlavenderblush[39m
[38;2;230;230;250mlavender[39m
[38;2;124;252;0mlawngreen[39m
[38;2;255;250;205mlemonchiffon[39m
[38;2;173;216;230mlightblue[39m
[38;2;240;128;128mlightcoral[39m
[38;2;224;255;255mlightcyan[39m
[38;2;250;250;210mlightgoldenrodyellow[39m
[38;2;211;211;211mlightgray[39m
[38;2;144;238;144mlightgreen[39m
[38;2;211;211;211mlightgrey[39m
[38;2;255;182;193mlightpink[39m
[38;2;255;160;122mlightsalmon[39m
[38;2;32;178;170mlightseagreen[39m
[38;2;135;206;250mlightskyblue[39m
[38;2;119;136;153mlightslategray[39m
[38;2;119;136;153mlightslategrey[39m
[38;2;176;196;222mlightsteelblue[39m
[38;2;255;255;224mlightyellow[39m
[38;2;0;255;0mlime[39m
[38;2;50;205;50mlimegreen[39m
[38;2;250;240;230mlinen[39m
[38;2;255;0;255mmagenta[39m
[38;2;128;0;0mmaroon[39m
[38;2;102;205;170mmediumaquamarine[39m
[38;2;0;0;205mmediumblue[39m
[38;2;186;85;211mmediumorchid[39m
[38;2;147;112;219mmediumpurple[39m
[38;2;60;179;113mmediumseagreen[39m
[38;2;123;104;238mmediumslateblue[39m
[38;2;0;250;154mmediumspringgreen[39m
[38;2;72;209;204mmediumturquoise[39m
[38;2;199;21;133mmediumvioletred[39m
[38;2;25;25;112mmidnightblue[39m
[38;2;245;255;250mmintcream[39m
[38;2;255;228;225mmistyrose[39m
[38;2;255;228;181mmoccasin[39m
[38;2;255;222;173mnavajowhite[39m
[38;2;0;0;128mnavy[39m
[38;2;253;245;230moldlace[39m
[38;2;128;128;0molive[39m
[38;2;107;142;35molivedrab[39m
[38;2;255;165;0morange[39m
[38;2;255;69;0morangered[39m
[38;2;218;112;214morchid[39m
[38;2;238;232;170mpalegoldenrod[39m
[38;2;152;251;152mpalegreen[39m
[38;2;175;238;238mpaleturquoise[39m
[38;2;219;112;147mpalevioletred[39m
[38;2;255;239;213mpapayawhip[39m
[38;2;255;218;185mpeachpuff[39m
[38;2;205;133;63mperu[39m
[38;2;255;192;203mpink[39m
[38;2;221;160;221mplum[39m
[38;2;176;224;230mpowderblue[39m
[38;2;128;0;128mpurple[39m
[38;2;102;51;153mrebeccapurple[39m
[38;2;255;0;0mred[39m
[38;2;188;143;143mrosybrown[39m
[38;2;65;105;225mroyalblue[39m
[38;2;139;69;19msaddlebrown[39m
[38;2;250;128;114msalmon[39m
[38;2;244;164;96msandybrown[39m
[38;2;46;139;87mseagreen[39m
[38;2;255;245;238mseashell[39m
[38;2;160;82;45msienna[39m
[38;2;192;192;192msilver[39m
[38;2;135;206;235mskyblue[39m
[38;2;106;90;205mslateblue[39m
[38;2;112;128;144mslategray[39m
[38;2;112;128;144mslategrey[39m
[38;2;255;250;250msnow[39m
[38;2;0;255;127mspringgreen[39m
[38;2;70;130;180msteelblue[39m
[38;2;210;180;140mtan[39m
[38;2;0;128;128mteal[39m
[38;2;216;191;216mthistle[39m
[38;2;255;99;71mtomato[39m
[38;2;64;224;208mturquoise[39m
[38;2;238;130;238mviolet[39m
[38;2;245;222;179mwheat[39m
[38;2;255;255;255mwhite[39m
[38;2;245;245;245mwhitesmoke[39m
[38;2;255;255;0myellow[39m
[38;2;154;205;50myellowgreen[39m"
`;

exports[`use colorNames name with ansis.hex() > print all named truecolor 1`] = `
"[38;2;240;248;255maliceblue[39m
[38;2;250;235;215mantiquewhite[39m
[38;2;0;255;255maqua[39m
[38;2;127;255;212maquamarine[39m
[38;2;240;255;255mazure[39m
[38;2;245;245;220mbeige[39m
[38;2;255;228;196mbisque[39m
[38;2;0;0;0mblack[39m
[38;2;255;235;205mblanchedalmond[39m
[38;2;0;0;255mblue[39m
[38;2;138;43;226mblueviolet[39m
[38;2;165;42;42mbrown[39m
[38;2;222;184;135mburlywood[39m
[38;2;95;158;160mcadetblue[39m
[38;2;127;255;0mchartreuse[39m
[38;2;210;105;30mchocolate[39m
[38;2;255;127;80mcoral[39m
[38;2;100;149;237mcornflowerblue[39m
[38;2;255;248;220mcornsilk[39m
[38;2;220;20;60mcrimson[39m
[38;2;0;255;255mcyan[39m
[38;2;0;0;139mdarkblue[39m
[38;2;0;139;139mdarkcyan[39m
[38;2;184;134;11mdarkgoldenrod[39m
[38;2;169;169;169mdarkgray[39m
[38;2;0;100;0mdarkgreen[39m
[38;2;169;169;169mdarkgrey[39m
[38;2;189;183;107mdarkkhaki[39m
[38;2;139;0;139mdarkmagenta[39m
[38;2;85;107;47mdarkolivegreen[39m
[38;2;255;140;0mdarkorange[39m
[38;2;153;50;204mdarkorchid[39m
[38;2;139;0;0mdarkred[39m
[38;2;233;150;122mdarksalmon[39m
[38;2;143;188;143mdarkseagreen[39m
[38;2;72;61;139mdarkslateblue[39m
[38;2;47;79;79mdarkslategray[39m
[38;2;47;79;79mdarkslategrey[39m
[38;2;0;206;209mdarkturquoise[39m
[38;2;148;0;211mdarkviolet[39m
[38;2;255;20;147mdeeppink[39m
[38;2;0;191;255mdeepskyblue[39m
[38;2;105;105;105mdimgray[39m
[38;2;105;105;105mdimgrey[39m
[38;2;30;144;255mdodgerblue[39m
[38;2;178;34;34mfirebrick[39m
[38;2;255;250;240mfloralwhite[39m
[38;2;34;139;34mforestgreen[39m
[38;2;255;0;255mfuchsia[39m
[38;2;220;220;220mgainsboro[39m
[38;2;248;248;255mghostwhite[39m
[38;2;218;165;32mgoldenrod[39m
[38;2;255;215;0mgold[39m
[38;2;128;128;128mgray[39m
[38;2;0;128;0mgreen[39m
[38;2;173;255;47mgreenyellow[39m
[38;2;128;128;128mgrey[39m
[38;2;240;255;240mhoneydew[39m
[38;2;255;105;180mhotpink[39m
[38;2;205;92;92mindianred[39m
[38;2;75;0;130mindigo[39m
[38;2;255;255;240mivory[39m
[38;2;240;230;140mkhaki[39m
[38;2;255;240;245mlavenderblush[39m
[38;2;230;230;250mlavender[39m
[38;2;124;252;0mlawngreen[39m
[38;2;255;250;205mlemonchiffon[39m
[38;2;173;216;230mlightblue[39m
[38;2;240;128;128mlightcoral[39m
[38;2;224;255;255mlightcyan[39m
[38;2;250;250;210mlightgoldenrodyellow[39m
[38;2;211;211;211mlightgray[39m
[38;2;144;238;144mlightgreen[39m
[38;2;211;211;211mlightgrey[39m
[38;2;255;182;193mlightpink[39m
[38;2;255;160;122mlightsalmon[39m
[38;2;32;178;170mlightseagreen[39m
[38;2;135;206;250mlightskyblue[39m
[38;2;119;136;153mlightslategray[39m
[38;2;119;136;153mlightslategrey[39m
[38;2;176;196;222mlightsteelblue[39m
[38;2;255;255;224mlightyellow[39m
[38;2;0;255;0mlime[39m
[38;2;50;205;50mlimegreen[39m
[38;2;250;240;230mlinen[39m
[38;2;255;0;255mmagenta[39m
[38;2;128;0;0mmaroon[39m
[38;2;102;205;170mmediumaquamarine[39m
[38;2;0;0;205mmediumblue[39m
[38;2;186;85;211mmediumorchid[39m
[38;2;147;112;219mmediumpurple[39m
[38;2;60;179;113mmediumseagreen[39m
[38;2;123;104;238mmediumslateblue[39m
[38;2;0;250;154mmediumspringgreen[39m
[38;2;72;209;204mmediumturquoise[39m
[38;2;199;21;133mmediumvioletred[39m
[38;2;25;25;112mmidnightblue[39m
[38;2;245;255;250mmintcream[39m
[38;2;255;228;225mmistyrose[39m
[38;2;255;228;181mmoccasin[39m
[38;2;255;222;173mnavajowhite[39m
[38;2;0;0;128mnavy[39m
[38;2;253;245;230moldlace[39m
[38;2;128;128;0molive[39m
[38;2;107;142;35molivedrab[39m
[38;2;255;165;0morange[39m
[38;2;255;69;0morangered[39m
[38;2;218;112;214morchid[39m
[38;2;238;232;170mpalegoldenrod[39m
[38;2;152;251;152mpalegreen[39m
[38;2;175;238;238mpaleturquoise[39m
[38;2;219;112;147mpalevioletred[39m
[38;2;255;239;213mpapayawhip[39m
[38;2;255;218;185mpeachpuff[39m
[38;2;205;133;63mperu[39m
[38;2;255;192;203mpink[39m
[38;2;221;160;221mplum[39m
[38;2;176;224;230mpowderblue[39m
[38;2;128;0;128mpurple[39m
[38;2;102;51;153mrebeccapurple[39m
[38;2;255;0;0mred[39m
[38;2;188;143;143mrosybrown[39m
[38;2;65;105;225mroyalblue[39m
[38;2;139;69;19msaddlebrown[39m
[38;2;250;128;114msalmon[39m
[38;2;244;164;96msandybrown[39m
[38;2;46;139;87mseagreen[39m
[38;2;255;245;238mseashell[39m
[38;2;160;82;45msienna[39m
[38;2;192;192;192msilver[39m
[38;2;135;206;235mskyblue[39m
[38;2;106;90;205mslateblue[39m
[38;2;112;128;144mslategray[39m
[38;2;112;128;144mslategrey[39m
[38;2;255;250;250msnow[39m
[38;2;0;255;127mspringgreen[39m
[38;2;70;130;180msteelblue[39m
[38;2;210;180;140mtan[39m
[38;2;0;128;128mteal[39m
[38;2;216;191;216mthistle[39m
[38;2;255;99;71mtomato[39m
[38;2;64;224;208mturquoise[39m
[38;2;238;130;238mviolet[39m
[38;2;245;222;179mwheat[39m
[38;2;255;255;255mwhite[39m
[38;2;245;245;245mwhitesmoke[39m
[38;2;255;255;0myellow[39m
[38;2;154;205;50myellowgreen[39m"
`;
